### PR TITLE
feat(app_icon): add app icon

### PR DESCRIPTION
**Pull request by aqupriyanka**
## Description
This PR takes care of adding the capability of url based App Icons

`<app-app-icon> </app-app-icon>`

Prerequisites:
Add VantageAppModule

### What's included?
- App Icons can be dispalyed using the component `<app-app-icon>`
- Need to import the VantageAppModule.

###Test Steps
- Update the proxy.conf.js server url to appcenter ux url
 - Navigate to localhost:4200
- Log in using user1 user.
 - Click Log In button

image

### CODE REVIEW

File: src/docs-ui/app/main/main.component.ts [hunk 1]
```diff
@@ -4,6 +4,8 @@ import { VantageToastService, VantageErrorService } from '@td-vantage/ui-platfor
 import { VantageSessionService } from '@td-vantage/ui-platform/auth';
 import { IUser } from '@td-vantage/ui-platform/user';
 import { timeout } from 'rxjs/operators';
+import { IApp, VantageAppsService } from '@td-vantage/ui-platform/app';
```
Comment (emoralesb05) [inline src/docs-ui/app/main/main.component.ts]: I know its not part of this PR, but can we rename this to `VantageAppService`?

File: src/lib/app/app-icon/app-icon.component.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import { Component, OnInit, Input } from '@angular/core';
+import { coerceBooleanProperty } from '@angular/cdk/coercion';
+import { IApp } from '..';
+
+@Component({
+  selector: 'app-app-icon',
```
Comment (emoralesb05) [inline src/lib/app/app-icon/app-icon.component.ts]: Can you rename this selector to `vui-app-icon` please?

File: src/lib/app/apps.service.ts [hunk 3]
```diff
@@ -53,6 +53,7 @@ export interface IApp {
   system_id?: string;
   containerPort?: number;
   favorite?: boolean;
+  isIconUrl?: boolean;
```
Comment (emoralesb05) [inline src/lib/app/apps.service.ts]: We shouldnt add this here since its not actually part of the App contract. Lets have a small function that does this inside the component.

File: src/lib/app/app-icon/app-icon.component.html [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<mat-icon
+  [ngClass]="_matListAvatar ? 'mat-list-avatar mat-icon-center' : ''"
+  [class]="color"
+  *ngIf="!app?.isIconUrl; else displayImgIcon"
```
Comment (emoralesb05) [inline src/lib/app/app-icon/app-icon.component.html]: Lets use `isUrl` here.

File: src/lib/app/app-icon/app-icon.component.ts (hunk 2 above)
Comment (emoralesb05) [reply src/lib/app/app-icon/app-icon.component.ts]: or maybe `vtd-`? haha

File: src/lib/app/app-icon/app-icon.component.ts [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    this._matListAvatar = coerceBooleanProperty(matListAvatar);
+  }
+  get active(): boolean {
+    return this._matListAvatar;
+  }
+
+  @Input() set app(app: IApp) {
+    this._app = app;
+    if (this._app && this._app.icon) {
+      this._app.isIconUrl = !!this.isUrl(this._app.icon);
```
Comment (christianmemije) [inline src/lib/app/app-icon/app-icon.component.ts]: This input should be read only, we should not be modifying this object.

File: src/lib/app/app-icon/app-icon.component.ts [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+  @Input('matListAvatar')
+  set matListAvatar(matListAvatar: boolean) {
+    this._matListAvatar = coerceBooleanProperty(matListAvatar);
+  }
+  get active(): boolean {
+    return this._matListAvatar;
+  }
+
+  @Input() set app(app: IApp) {
```
Comment (christianmemije) [inline src/lib/app/app-icon/app-icon.component.ts]: No need for a setter and getter for app. `isIconUrl` can just be a getter.

File: src/lib/app/app-icon/app-icon.component.ts [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+import { IApp } from '..';
+
+@Component({
+  selector: 'app-app-icon',
+  templateUrl: './app-icon.component.html',
+  styleUrls: ['./app-icon.component.scss'],
+})
+export class VantageAppIconComponent {
+  private _app: IApp;
+  _matListAvatar: boolean;
```
Comment (christianmemije) [inline src/lib/app/app-icon/app-icon.component.ts]: Should default to false. And should only be prefixed with an underscore if it is private.

File: src/docs-ui/app/main/main.component.ts (hunk 1 above)
Comment (JulieMarie) [reply src/docs-ui/app/main/main.component.ts]: I went back and forth on whether to name it VantageAppService or VantageAppsService.  Should she rename  the files to `app.service.spec.ts` and `app.service.ts` too?

File: src/lib/app/app-icon/app-icon.component.ts (hunk 2 above)
Comment (JulieMarie) [reply src/lib/app/app-icon/app-icon.component.ts]: `vui-app-icon` lol

**Comment by jeremysmartt** [comment]
closing this PR, OBE
